### PR TITLE
Update C# docs with latest .NET Core info

### PR DESCRIPTION
**Pull request by akoeplinger**
We can close a bunch of old issues with this:

Fixes https://github.com/travis-ci/travis-ci/issues/7301
Fixes https://github.com/travis-ci/travis-ci/issues/7255
Fixes https://github.com/travis-ci/travis-ci/issues/7026
Fixes https://github.com/travis-ci/travis-ci/issues/4663

This one was done a while ago:

Fixes https://github.com/travis-ci/travis-ci/issues/5189

@BanzaiMan 

/cc @Joshua-Anderson @nterry

### CODE REVIEW

File: user/languages/csharp.md [hunk 1]
```diff
@@ -73,46 +75,46 @@ mono:
...
 You can choose from the following Mono versions:
 
-| Version          | Installed Packages (linux only, OSX always includes everything)  |
+| Version          | Installed Packages (Linux only, OSX always includes everything)  |
 | ---------------- | ---------------------------------------------------------------- |
 | 3.10.0 and later | mono-complete, mono-vbnc, fsharp, nuget, referenceassemblies-pcl |
 | 3.8.0            | mono-complete, mono-vbnc, fsharp, nuget                          |
 | 3.2.8            | mono-complete, mono-vbnc, fsharp                                 |
 | 2.10.8           | mono-complete, mono-vbnc                                         |
+| none             |                                                                  |
```
Comment (plaindocs) [inline user/languages/csharp.md]: What I am missing that makes this line relevant?

File: user/languages/csharp.md [hunk 2]
```diff
@@ -73,46 +75,46 @@ mono:
...
 ```yml
 language: csharp
 mono: none
-dotnet: 1.0.0-preview2-003121
+dotnet: 1.0.1
 ...
 ```
 
-You can choose from the following .NET Core versions:
+*Note*: you need to specify the version number of the .NET Core SDK (_not_ the .NET Core Runtime). For example, the .NET Core SDK 1.0.1 contains both the .NET Core Runtime 1.0.4 and 1.1.1.
```
Comment (plaindocs) [inline user/languages/csharp.md]: You can highlight notes with markdown `>` quotes

> like this

**Comment by plaindocs** [comment]
This all looks fine to me, although the topic itself could use a rewrite, but better to merge these in than not I think.

File: user/languages/csharp.md (hunk 1 above)
Comment (akoeplinger) [inline user/languages/csharp.md]: you can turn off Mono installation with this (e.g. if you only want .NET Core). I added text to clarify.

**Comment by plaindocs** [comment]
@BanzaiMan anything further to add?

**Pull request merged**